tests/SchedulerTest.php

```diff
@@ -114,14 +114,13 @@ function (Task $_, array $promises) use (&$canceledPromises): void {
                 }
             );
 
-        // Tick callbacks are guaranteed to be executed in the order they are enqueued,
-        // thus we will not be able to remove the task before it is finished. Therefore,
-        // we need to use timers with 0 intervals.
-        Loop::addTimer(0, function () use ($task): void {
+        Loop::addTimer(.001, function () use ($task): void {
```

Comment:
Please explain, why we're sleeping for `.001` seconds. This change should also be in a separate commit.